ndcube/visualization/descriptor.py

```diff
@@ -18,35 +21,60 @@ def __set_name__(self, owner, name):
         # attribute name is the name of the attribute on the parent class where
         # the data is stored.
         self._attribute_name = f"_{name}"
+        plotter = self._resolve_default_type(raise_error=False)
+        if plotter is not None and hasattr(plotter, "plot"):
+            functools.update_wrapper(owner.plot, plotter.plot)
 
-    def __get__(self, obj, objtype=None):
-        if obj is None:
-            return
-
-        if getattr(obj, self._attribute_name, None) is None:
-
-            # We special case the default MatplotlibPlotter so that we can
-            # delay the import of matplotlib until the plotter is first
-            # accessed.
+    def _resolve_default_type(self, raise_error=True):
+        # We special case the default MatplotlibPlotter so that we can
+        # delay the import of matplotlib until the plotter is first
+        # accessed.
+        if self._default_type in ("mpl_plotter", "mpl_sequence_plotter"):
             try:
                 if self._default_type == "mpl_plotter":
                     from ndcube.visualization.mpl_plotter import MatplotlibPlotter
-                    self.__set__(obj, MatplotlibPlotter)
+                    return MatplotlibPlotter
                 elif self._default_type == "mpl_sequence_plotter":
                     from ndcube.visualization.mpl_sequence_plotter import MatplotlibSequencePlotter
-                    self.__set__(obj, MatplotlibSequencePlotter)
-                elif self._default_type is not None:
-                    self.__set__(obj, self._default_type)
-                else:
-                    # If we have no default type then just return None
-                    return
+                    return MatplotlibSequencePlotter
             except ImportError as e:
-                raise ImportError(MISSING_MATPLOTLIB_ERROR_MSG) from e
+                if raise_error:
+                    raise ImportError(MISSING_MATPLOTLIB_ERROR_MSG) from e
+                else:
+                    warnings.warn(MISSING_MATPLOTLIB_ERROR_MSG)
```

Comment:
I wouldn't even bother doing this here. I would just pass silently, we will error if and when they call `.plot()` or access `.plotter`.